s3path.py

```diff
@@ -920,3 +986,28 @@ def is_symlink(self, *args, **kwargs):
 
     def stat(self):
         return self._stat
+
+
+def register_uri_pathlib_s3_backend():
+    """Register s3:// pathlib implementation
+
+    return a tuple with
+    (scheme, PurePathClass, PathClass, Optional[params_adapter], )
+    """
+    return ("s3", PureS3Path, S3Path, None)
+
+
+original_path_rename = Path.rename
+
+
+def __mokey_patch_path_rename__(self, target):
+    if isinstance(target, str):
+        target = PathFactory(target)
+    if self.__class__ is S3Path and target.__class__ is not S3Path:
+        return self._rename_to_fs(target)
+    elif self.__class__ is not S3Path and target.__class__ is S3Path:
+        return target._rename_from_fs(self)
+    return original_path_rename(self, target)
+
+
+Path.rename = __mokey_patch_path_rename__
```

Comment:
I'll thinking to move this patch on uri-pathlib-factory library somehow

Reply:
What's could be great to achieved is something lire this (Assuming ftp backend is developed)

```python
Path("ftp://path/to/file.txt").rename("s3://buckekt/path/to/s3-file.txt")
```

In order to move files from any uri backend !